app/src/main/java/com/android/streetworkapp/ui/map/MapUi.kt

```diff
@@ -169,6 +188,84 @@ fun MapScreen(
               }
         }
   }
+  // Settings variable defined beforehand :
+  // Affected by the filter settings => changes confirmed when confirming (onSubmit).
+
+  // Display the Filter component :
+  CustomDialog(
+      showFilterSettings,
+      tag = "Filter",
+      dialogType = DialogType.CONFIRM,
+      title = LocalContext.current.getString(R.string.park_filter_title),
+      Content = { ParkFilterSettings(userFilterInput) },
+      onSubmit = { filter.set(userFilterInput) },
+      onDismiss = { userFilterInput.set(filter) })
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ParkFilterSettings(userFilterInput: FilterSettings) {
+  val context = LocalContext.current
+
+  // Note - This is a composable and cannot be defined in the ColorPalette (Theme.kt).
+  val filterChipColors =
+      FilterChipDefaults.filterChipColors(
+          selectedLabelColor = ColorPalette.PRINCIPLE_BACKGROUND_COLOR,
+          selectedContainerColor = ColorPalette.INTERACTION_COLOR_DARK)
+
+  Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
+    // Park rating filter :
+    Text(
+        text =
+            context.getString(
+                R.string.rating_filter_title, userFilterInput.minRating.value.toString()),
+        fontSize = Type.bodyLarge.fontSize,
+        modifier = Modifier.testTag("ratingFilterTitle"))
+    InteractiveRatingComponent(userFilterInput.minRating)
+
+    HorizontalDivider()
+
+    // Event quantity filter :
+    Text(
+        context.getString(R.string.eventDensity_filter_title),
+        fontSize = Type.bodyLarge.fontSize,
+        modifier = Modifier.testTag("eventDensityFilterTitle"))
+
+    Row(modifier = Modifier.align(Alignment.CenterHorizontally).fillMaxWidth(0.825f)) {
+      FilterChip(
+          selected = userFilterInput.eventDensity.contains(EventDensity.LOW),
+          onClick = { userFilterInput.updateDensity(EventDensity.LOW) },
+          label = { Text(EventDensity.LOW.name) },
+          colors = filterChipColors,
+          modifier = Modifier.padding(end = 2.dp).testTag("lowDensityFilterChip"))
+
+      FilterChip(
+          selected = userFilterInput.eventDensity.contains(EventDensity.MEDIUM),
+          onClick = { userFilterInput.updateDensity(EventDensity.MEDIUM) },
+          label = { Text(EventDensity.MEDIUM.name) },
+          colors = filterChipColors,
+          modifier = Modifier.padding(end = 2.dp).testTag("mediumDensityFilterChip"))
+
+      FilterChip(
+          selected = userFilterInput.eventDensity.contains(EventDensity.HIGH),
+          onClick = { userFilterInput.updateDensity(EventDensity.HIGH) },
+          label = { Text(EventDensity.HIGH.name) },
```

Comment:
For all three filterChips, putting LOW, MEDIUM, HIGH as the labels feels a bit vague for users to read, it might be nice to put the specific values as labels instead to help users better choose the density, for example: LOW (0 < ... < 2) or just (0 < ... < 2)

Reply:
I propose something in between as the ranges take too much space (reduced width). I came up with the following :

<p align="center">
<img src="https://github.com/user-attachments/assets/1ca6f9a9-dc1c-4ab1-b303-4a317cf34cec" alt="image" width="195">
</p>

Do you agree with such display ? It no longer give abstracted types but a direct idea of the range.

Reply:
This is great!